src/components/Analytics/detailedCharts/answersDistribution/AnswersDistributionSlider.js

```diff
@@ -0,0 +1,62 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import AnswersDistributionBarChart from './AnswersDistributionBarChart';
+
+const AnswersDistributionSlider = ({
+  maxWidth,
+  question,
+  appDataForQuestion,
+}) => {
+  const getResponsesByValue = useCallback(() => {
+    return appDataForQuestion.groupBy((r) => r.data.value);
+  }, [appDataForQuestion]);
+
+  const [responsesByValue, setResponsesByValue] = useState(
+    getResponsesByValue()
+  );
+
+  useEffect(
+    () => setResponsesByValue(getResponsesByValue()),
+    [getResponsesByValue]
+  );
+
+  const chartData = Array.from(responsesByValue)
+    .sort(([val1], [val2]) => {
+      if (val1 < val2) {
+        return -1;
+      } else if (val1 > val2) {
+        return 1;
+      } else {
+        return 0;
+      }
+    })
+    .reduce(
+      (acc, [value, list], idx) => {
+        return {
+          data: {
+            x: [...acc.data.x, value],
+            y: [...acc.data.y, list.size],
+          },
+          percentage: [...acc.percentage, list.size / appDataForQuestion.size],
+          maxValue: Math.max(acc.maxValue, list.size),
+          hoverText: [...acc.hoverText, list.size],
+        };
+      },
+      {
+        data: { x: [], y: [] },
+        percentage: [],
+        maxValue: 0,
+        hoverText: [],
+      }
+    );
```

Comment:
I think you've written this often, could this be factor out?

Reply:
I thought about it, but wasn't sure about how to do it properly. I will try to think more about it, thanks.